projector-launcher/src/main/kotlin/ElectronApp.kt

```diff
@@ -213,6 +201,18 @@ class ElectronApp(val url: String) {
           require("open")(url)
         }
       })
+
+      contents.on("before-input-event", listener = { e: Event, input: Input ->
+        if (process.platform.equals("darwin")) {
+          if (input.key === "Q" && !input.control && !input.alt && input.meta && !input.shift) {
+            this.quitApp()
+          }
+        } else {
+          if (input.key === "F4" && !input.control && input.alt && !input.meta && !input.shift) {
+            this.quitApp()
+          }
+        }
+      })
```

Comment:
Great PR, but there is a small issue.

I've tested and found a case when it doesn't work. Please see comment in the proposed change. Also, I've moved the lambda outside of the `on` call just to follow code style. And marked `Event` as unused.

```suggestion
      contents.on("before-input-event") handleExitHotkey@ { _: Event, input: Input ->
        if (input.type != "keyDown") {
          // For every input, two events can be generated: keyDown and keyUp.
          // When we close an app above our electron app, keyUp can be sent to our app because
          // the hotkey can be released after the app above is closed.
          // So need to react only to keyDown event.
          return@handleExitHotkey
        }

        if (process.platform.equals("darwin")) {
          if (input.key === "Q" && !input.control && !input.alt && input.meta && !input.shift) {
            this.quitApp()
          }
        } else {
          if (input.key === "F4" && !input.control && input.alt && !input.meta && !input.shift) {
            this.quitApp()
          }
        }
      }
```

Please accept the change and I will merge the PR.

Reply:
Oh, actually, the accept button's been available for me, so I've done it myself :) Will merge now. Thanks for the PR!

Reply:
Oh awesome - just logged on for the day. Thank you very much!